nucleus/security/core/src/main/java/com/sun/enterprise/security/GroupPrincipal.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2021 Contributors to Eclipse Foundation. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package com.sun.enterprise.security;
+
+import java.util.Enumeration;
+import java.security.Principal;
+
+/**
+ * A group of principals.
+ * 
+ * @author Arjan Tijms
+ *
+ */
+public interface GroupPrincipal extends Principal {
```

Comment:
[org.glassfish.security.common.Group](https://github.com/eclipse-ee4j/glassfish/blob/master/nucleus/common/common-util/src/main/java/org/glassfish/security/common/Group.java#L26) can be used as replacement.

Reply:
I considered this, but org.glassfish.security.common.Group is not an interface and drags along other details from its base class that doesn't make it an ideal replacement.